types/src/account_config/events/coin_register_event.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::account_config::TypeInfo;
+use anyhow::Result;
+use move_core_types::{
+    ident_str,
+    identifier::IdentStr,
+    language_storage::{StructTag, TypeTag, CORE_CODE_ADDRESS},
+    move_resource::MoveStructType,
+};
+use once_cell::sync::Lazy;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CoinRegisterEvent {
+    type_info: TypeInfo,
+}
+
+impl CoinRegisterEvent {
+    pub fn new(type_info: TypeInfo) -> Self {
+        Self { type_info }
+    }
+
+    pub fn try_from_bytes(bytes: &[u8]) -> Result<Self> {
+        bcs::from_bytes(bytes).map_err(Into::into)
+    }
+}
+
+impl MoveStructType for CoinRegisterEvent {
+    const MODULE_NAME: &'static IdentStr = ident_str!("account");
+    const STRUCT_NAME: &'static IdentStr = ident_str!("CoinRegisterEvent");
+}
+
+pub const COIN_REGISTER_EVENT_TYPE_STR: &str =
```

Comment:
I prefer this can be generated but not hardcoded.

Reply:
Removed these constants.